pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MultiStageOperator.java

```diff
@@ -223,6 +232,25 @@ protected Map<String, Plan.ExplainNode.AttributeValue> getExplainAttributes() {
     return Collections.emptyMap();
   }
 
+  private long getGcTimeMillis() {
```

Comment:
I think the idea is good but mixing all gc collection times might make it useless . When I run the code as test I see the following beans :
- g1 young generation
- g1 concurrent gc 
- g1 old generation 
I'm afraid metric it's going to be dominated by concurrent gc (which should be ignored). Ideally, we'd be able to use metric to learn e.g. how long operator was blocked due to stop-the-world gc events, but api doesn't really allow it.

Reply:
I also thought about that, but as you said there is no much we can do. On the other hand:
1. If we end up having full gcs due to the query, this may be useful (although GCs _produced_ by a stage will also affect other stages on the same machine).
2. Even if concurrent GC is dominant... that is a useful right? Even if we don't stop the world, during that time we had less CPUs available for the program. Again, the stat on one stage will be affected by others, but it may be better than nothing. Given it is disabled by default, I think it is fine

Reply:
I do agree to some extent but still believe that there'll be too much noise for this stat to be useful (vs checking gc stats in grafana). Maybe it'd be better to use jfr api and capture select stats.